runtime/interpreter/value.go

```diff
@@ -16486,6 +16486,8 @@ func (v *DictionaryValue) IsResourceKinded(interpreter *Interpreter) bool {
 type OptionalValue interface {
 	Value
 	isOptionalValue()
+	iter(f func(Value))
+	fmap(inter *Interpreter, f func(Value) Value) OptionalValue
```

Comment:
now we gotta extend optionals all the way with `<*>`, `>>=`, `<|>`, and `<>`

Reply:
I would love to make `OptionalValue` generic in its `Value` (i.e. something like `type OptionalValue[T any] interface { ...` so that less downcasting is necessary in cases like these, but it seems like a lot of restructuring of code in order to make Go not complain about circular references in the visitor.